### PR TITLE
Okta v2

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/18369

## Description
Refactored Okta integration - convert the javascript code into python 
[Integration Enhancement] Okta - Add command: "Clear User Sessions"
## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Integrations/OktaV2/OktaV2.yml [hunk 1]
```diff
@@ -0,0 +1,1790 @@
...
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+description: Integration with Okta's cloud-based identity management service
+display: OktaV2
+name: OktaV2
```
Comment (Itay4) [inline Integrations/OktaV2/OktaV2.yml]:
```suggestion
name: Okta v2
```

File: Integrations/OktaV2/OktaV2.yml [hunk 2]
```diff
@@ -0,0 +1,1790 @@
...
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+description: Integration with Okta's cloud-based identity management service
+display: OktaV2
```
Comment (Itay4) [inline Integrations/OktaV2/OktaV2.yml]:
```suggestion
display: Okta v2
```

File: Integrations/OktaV2/OktaV2.yml [hunk 3]
```diff
@@ -0,0 +1,1790 @@
+category: Authentication
+commonfields:
+  id: OktaV2
```
Comment (Itay4) [inline Integrations/OktaV2/OktaV2.yml]:
```suggestion
  id: Okta v2
```

File: Integrations/OktaV2/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+
+
+## [19.11.1] - 2019-11-26
+These notes will be published in the next release notes
```
Comment (Itay4) [inline Integrations/OktaV2/CHANGELOG.md]: whats with the rn?

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 5]
```diff
@@ -0,0 +1,1007 @@
...
+    Args:
+        client: HelloWorld client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    uri = 'users/me'
+    client._http_request(method='GET', url_suffix=uri)
+    return demisto.results('ok')
+    raise ValueError('Test OktaV2 integration failed')
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: unreachable code

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 6]
```diff
@@ -0,0 +1,1007 @@
...
+
+
+# TODO: Manage the response codes
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: ?

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 7]
```diff
@@ -0,0 +1,1007 @@
...
+
+def get_group_members_command(client, args):
+    if not (args.get('groupId') or args.get('groupName')):
+        return_error("Missing arguments for command")
+    limit = args.get('limit')
+    group_id = args.get('groupId') or client.get_group_id(args.get('groupName'))
+    raw_members = client.get_group_members(group_id, limit)
+    # group = client.get_readable_groups(client.get_groups(args))
+    # for member in raw_members:
+    #     member['group'] = group
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: ?

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 8]
```diff
@@ -0,0 +1,1007 @@
...
+    return (
+        readable_output,
+        outputs,
+        raw_response
+    )
+
+
+def get_logs_command(client, args):
+    raw_response = client.get_logs(args)
+    if len(raw_response) == 0:
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]:
```suggestion
    if not raw_response:
```

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 9]
```diff
@@ -0,0 +1,1007 @@
...
+
+def get_group_assignments_command(client, args):
+    args['filter'] = 'eventType eq "group.user_membership.add"'
+    raw_response = client.get_logs(args)
+    if len(raw_response) == 0:
+        return 'No logs found', {}, raw_response
+    logs = client.get_readable_logs(raw_response)
+    readable_output = tableToMarkdown('Group Assignment Events', logs)
+    outputs = {
+        'Okta.Logs.Events(val.uuid===obj.uuid)': createContext(raw_response)
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]:
```suggestion
        'Okta.Logs.Events(val.uuid && val.uuid === obj.uuid)': createContext(raw_response)
```

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 10]
```diff
@@ -0,0 +1,1007 @@
+from CommonServerPython import *
+
+# IMPORTS
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+# CONSTANTS
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+SEARCH_LIMIT = 200
+# TODO: Ask Shai whether to keep it as a dict or create a class (user class, group class etc )
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: leave as is, remove comment

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py (hunk 6 above)
Comment (Bargenish) [reply Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: already managed it and forgot to delete the comment

File: Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py [hunk 11]
```diff
@@ -914,7 +928,7 @@ def get_application_authentication_command(client, args):
 
 def delete_user_command(client, args):
     if not (args.get('username') or args.get('userId')):
-        return_error("Missing arguments for command")
+        raise Exception("Missing arguments for command")
```
Comment (yaakovi) [inline Packs/Okta_v2/Integrations/Okta_v2/Okta_v2.py]: improve error message: `You must provide either username or userId`

**Comment by kirbles19** [comment]
@Bargenish Done. I sent you the completed YML in a DM in Slack because there are more than 5k changed files, not sure where to add it.

**Pull request closed without merging**